test/powershell/Modules/Microsoft.PowerShell.Management/Test-Path.Tests.ps1

```diff
@@ -2,123 +2,136 @@
 
-	# populate with additional files
-	New-Item -Path $testdirectory -Name datestfile -value 1 -ItemType file | Out-Null
-	New-Item -Path $testdirectory -Name gatestfile -value 1 -ItemType file | Out-Null
-	New-Item -Path $testdirectory -Name usr -value 1 -ItemType directory | Out-Null
+        # populate with additional files
+        New-Item -Path $testdirectory -Name datestfile -value 1 -ItemType file | Out-Null
+        New-Item -Path $testdirectory -Name gatestfile -value 1 -ItemType file | Out-Null
+        New-Item -Path $testdirectory -Name usr -value 1 -ItemType directory | Out-Null
 
-	$nonExistentDir = Join-Path -Path (Join-Path -Path $testdirectory -ChildPath usr) -ChildPath bin
-	$nonExistentPath = Join-Path -Path (Join-Path -Path (Join-Path -Path $testdirectory -ChildPath usr) -ChildPath bin) -ChildPath error
+        $nonExistentDir = Join-Path -Path (Join-Path -Path $testdirectory -ChildPath usr) -ChildPath bin
+        $nonExistentPath = Join-Path -Path (Join-Path -Path (Join-Path -Path $testdirectory -ChildPath usr) -ChildPath bin) -ChildPath error
     }
 
     It "Should be called on an existing path without error" {
-	{ Test-Path $testdirectory }              | Should -Not -Throw
-	{ Test-Path -Path $testdirectory }        | Should -Not -Throw
-	{ Test-Path -LiteralPath $testdirectory } | Should -Not -Throw
+        { Test-Path $testdirectory }              | Should -Not -Throw
+        { Test-Path -Path $testdirectory }        | Should -Not -Throw
+        { Test-Path -LiteralPath $testdirectory } | Should -Not -Throw
     }
 
     It "Should allow piping objects to it" {
-	{ $testdirectory | Test-Path  } | Should -Not -Throw
+        { $testdirectory | Test-Path  } | Should -Not -Throw
 
-	$testdirectory                  | Test-Path | Should -BeTrue
-	$nonExistentDir                 | Test-Path | Should -BeFalse
+        $testdirectory                  | Test-Path | Should -BeTrue
+        $nonExistentDir                 | Test-Path | Should -BeFalse
     }
 
     It "Should be called on a nonexistent path without error" {
-	{ Test-Path -Path $nonExistentPath } | Should -Not -Throw
+        { Test-Path -Path $nonExistentPath } | Should -Not -Throw
     }
 
     It "Should return false for a nonexistent path" {
-	Test-Path -Path $nonExistentPath | Should -BeFalse
+        Test-Path -Path $nonExistentPath | Should -BeFalse
     }
 
     It "Should return true for an existing path" {
-	{ Test-Path -Path $testdirectory } | Should -BeTrue
+        { Test-Path -Path $testdirectory } | Should -BeTrue
+    }
+
+    It 'Should return false for an empty string' {
+        Test-Path -Path '' | Should -BeFalse
+    }
+
+    It 'Should return false for a whitespace string' {
+        Test-Path -Path '  ' | Should -BeFalse
+    }
+
+    It 'Should write a non-terminating error when given a null path' {
+        { Test-Path -Path $null -ErrorAction Stop }             | Should -Throw -ErrorId 'NullPathNotPermitted'
```

Comment:
I believe we should tests here `Test-Path $null, $null;`.

Reply:
Done. Also had to modify some `ErrorPosition` tests as they were causing infinite loops when the cmldet simply wrote a non-terminating error.